testing/integration/upgrade_test.go

```diff
@@ -208,11 +211,56 @@ func TestStandaloneUpgrade(t *testing.T) {
 
 			parsedUpgradeVersion, err := version.ParseVersion(define.Version())
 			require.NoErrorf(t, err, "define.Version() %q cannot be parsed as agent version", define.Version())
-			testStandaloneUpgrade(ctx, t, agentFixture, parsedVersion, parsedUpgradeVersion, "", true)
+			skipVerify := version_8_7_0.Less(*parsedVersion)
+			testStandaloneUpgrade(ctx, t, agentFixture, parsedVersion, parsedUpgradeVersion, "", skipVerify, true, false, "")
 		})
 	}
 }
 
+func TestStandaloneUpgradeWithGPGFallback(t *testing.T) {
+	define.Require(t, define.Requirements{
+		Local: false, // requires Agent installation
+		Sudo:  true,  // requires Agent installation
+	})
+
+	minVersion := version_8_10_0_SNAPSHOT
+	parsedVersion, err := version.ParseVersion(define.Version())
+	require.NoError(t, err)
+
+	if parsedVersion.Less(*minVersion) {
+		t.Skipf("Version %s is lower than min version %s", define.Version(), minVersion)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// previous
+	toVersion, err := parsedVersion.GetPreviousMinor()
```

Comment:
Not the biggest fan of trying to guess the previous minor by decreasing the counters...
Since we already have access to artifact api you could retrieve the previous version string from there...
Not a blocker in itself until we release 9.x it's just that we can rely on the actual agent versions...